pkgs/games/unvanquished/default.nix

```diff
@@ -0,0 +1,156 @@
+      interpreter="$(< "$NIX_CC/nix-support/dynamic-linker")"
+      for f in pnacl/bin/*; do
+        if [ -f "$f" && -x "$f" ]; then
+          echo "Patching $f"
+          patchelf --set-interpreter "$interpreter" "$f"
+        fi
+      done
+    '';
+    preCheck = "pnacl/bin/clang -v"; # check it links correctly
+    installPhase = ''
+      mkdir -p $out
+      cp -R ./* $out/
+    '';
+  };
+
+  preload = stdenv.mkDerivation {
+    name = "nix-preload-for-unvanquished-nacl";
+    buildCommand = ''
+      mkdir $out/etc -p
+      echo ${gcc.cc.lib}/lib/libstdc++.so.6 > $out/etc/ld-nix.so.preload
+    '';
+  };
+
+  fhsEnv = buildFHSUserEnv {
+    name = "unvanquished-fhs-wrapper";
+    targetPkgs = pkgs: [ gcc.cc.lib preload ];
+  };
+
+  unvanquished-assets = stdenv.mkDerivation {
+    pname = "unvanquished-assets";
+    inherit version;
+
+    src = fetchurl {
+      # master as of 2021-04-09: the 0.51.2 script isn't as great
+      url = "https://raw.githubusercontent.com/Unvanquished/Unvanquished/2783e77acc2e41fc1f717124394ed851f1b612e6/download-paks";
+      sha256 = "1zbsi0yzfvgassg6dqks66qpjqa4yxi2ymis2qh8aj9dvkgqk6im";
+    };
+    outputHash = "sha256:15vndprms0vi5rbvf67hqxwgs3xhlcrxbzn4z3kz9d54g0z1zbzh";
+    outputHashMode = "recursive";
+    nativeBuildInputs = [ aria2 cacert ];
+    buildCommand = "bash $src --no-check $out";
+  };
+
+# this really is the daemon game engine, the game itself is in the assets
+in stdenv.mkDerivation rec {
+  name = "unvanquished-${version}";
+  inherit version binary-deps-version;
+
+  src = fetchgit {
+    url = "https://github.com/Unvanquished/Unvanquished.git";
+    rev = "v${version}";
+    sha256 = "1x943sxbgd41dhhc4j43y8vdwj74g9rbxyz1d7an82zbjsp34lak";
+  };
+  preConfigure = ''
+    mkdir daemon/external_deps/linux64-${binary-deps-version}/
+    cp -r ${unvanquished-binary-deps}/* daemon/external_deps/linux64-${binary-deps-version}/
+    chmod +w -R daemon/external_deps/linux64-${binary-deps-version}/
+  '';
+
+  nativeBuildInputs = [ cmake unvanquished-binary-deps copyDesktopItems ];
+  buildInputs = [
+    gmp
+    libGL
+    zlib
+    ncurses
+    geoip
+    lua5
+    nettle
+    curl
+    SDL2
+    freetype
+    glew
+    openal
+    libopus
+    opusfile
+    libogg
+    libvorbis
+    libjpeg
+    libwebp
+    libpng
+    libtheora # TODO: drop this one when 0.52 is released
+  ];
+
+  cmakeFlags = [
+    "-DBUILD_CGAME=NO"
+    "-DBUILD_SGAME=NO"
+    "-DUSE_HARDENING=TRUE"
+    "-DUSE_LTO=TRUE"
+  ];
+
+  desktopItem = makeDesktopItem {
+    name = "Unvanquished";
+    desktopName = "unvanquished";
+    exec = "daemon";
+    icon = "unvanquished";
+    categories = "Game;";
+  };
+
+  installPhase = ''
+    for f in daemon daemon-tty daemonded nacl_loader nacl_helper_bootstrap; do
+      install -Dm0755 -t $out/lib/ $f
+    done
+    install -Dm0644 -t $out/lib/ irt_core-x86_64.nexe
+
+    mkdir $out/bin/
+    cat > $out/bin/daemon-unwrapped <<-EOT
```

Comment:
Your `daemon-unwrapped` is wrapping a call to `lib/daemon` and is thus misleading. Maybe just call it `unvanquished` or `run-unvanquished-game`?

Reply:
I've reworked the "call stack" so that it is now

```
bin/daemon -> bin/daemon-no-fhs -> lib/daemon
bin/daemonded -> bin/daemonded-no-fhs ->lib/daemonded
bin/daemon-tty -> bin/daemon-tty-no-fhs -> lib/daemon-tty
```

Hopefully it is more clear. I'm not sure if things such as daemon-tty-no-fhs belong in bin/ though.

Reply:
For the record, this is now
```
bin/unvanquished -> lib/daemon-wrapper -> lib/daemon
bin/unvanquished-server -> lib/daemonded-wrapper ->lib/daemonded
bin/unvanquished-tty -> lib/daemon-tty-wrapper -> lib/daemon-tty
```